test/historical-query.spec.ts

```diff
@@ -40,7 +44,38 @@ let CONSEQUENCE: Consequence = {
 };
 
 describe("rules from AJO", () => {
-  it("supports historical condition for searchType ANY", () => {
+  const records = [
+    {
+      "iam.id":
+        "6cd5a8ed-e183-48b7-a0ef-657a4467df74#0477a309-6f63-4638-b729-ab51cf5dd3aa",
+      "iam.eventType": "display",
+      "iam.action": "",
+      someMagicalKey: 123,
+      futureKeys: "secret_abc_keys",
+      "ajo.id": 890,
+      firstTimestamp: 1609086720010,
+      timestamp: 1681321319855,
+    },
+    {
+      "iam.id":
+        "8cd5a8ed-e183-48b7-a0ef-657a4467df74#0477a309-6f63-4638-b729-ab51cf5dd3aa",
+      "iam.eventType": "interact",
+      "iam.action": "",
+      someMagicalKey: 123,
+      futureKeys: "secret_abc_keys",
+      "ajo.id": 890,
+      firstTimestamp: 1609086720010,
```

Comment:
I love that you added more keys here, i'd like to see tests that query on some of these and do not use `iam.eventType` nor `iam.id`.

such as 

```js
{
  "definition": {
    "events": [
      {
        "ajo.id": "890",
        "futureKeys": "secret_abc_keys"
      }
    ],
    "matcher": "ge",
    "value": 1
  },
  "type": "historical"
}

```

Reply:
done